modules/widget-visibility/widget-conditions.php

```diff
@@ -199,6 +213,19 @@ public static function widget_admin_setup() {
 		wp_localize_script( 'widget-conditions', 'widget_conditions_parent_pages', $all_parents );
 	}
 
+	public static function get_pages() {
+		global $wpdb;
+
+		$last_changed = wp_cache_get_last_changed( 'posts' );
+		$cache_key    = "get_pages:$last_changed";
+		$pages        = wp_cache_get( $cache_key, 'widget_conditions' );
+		if ( false === $pages ) {
+			$pages = $wpdb->get_results( "SELECT {$wpdb->posts}.ID, {$wpdb->posts}.post_parent, {$wpdb->posts}.post_title FROM {$wpdb->posts} WHERE {$wpdb->posts}.post_type = 'page' AND {$wpdb->posts}.post_status = 'publish' ORDER BY {$wpdb->posts}.post_date DESC" );
+			wp_cache_set( $cache_key, $pages, 'widget_conditions' );
+		}
+		return $pages;
```

Comment:
I wonder if we should add the `get_pages` filter in there, just in case folks are using the filter in their installs today, and would expect that filter to continue to have an impact on the Widget Visibility feature as well?

Reply:
Added. But, there's a concern I have. Since we are not returning full page object (the object we return only contains ID, post_title, and post_paren), and actually is not even a `WP_Post` object, there might be some confusion in the callback, where some of the properties are not present. But, since the `fields` argument can also limit the fields in the object in standard `get_pages` call, it might not really be an issue. Thoughts?

Reply:
Ah yes, you're right, I didn't think of that. I see [there are actually quite a few plugins that rely on the filter](https://wpdirectory.net/search/01EN3C8VR33KFVQHKX1CQXG1Y1), so it may be a bit risky for us to play with that filter here if some folks expect a `WP_Post`.

Maybe we skip using the filter for now, and I'll keep in mind if reports come up after the release. What do you think?

Reply:
Another approach would be to introduce a Jetpack specific filter and mention it in the release notes, so folks who want to take advantage of the filter in terms of the Widget Visibility feature, would have a path forward.

Reply:
Yeah, I like that idea! 👍

Reply:
Done ✅